sdk/internal/testframework/request_matcher.go

```diff
@@ -0,0 +1,111 @@
+// +build go1.13
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package testframework
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"reflect"
+
+	"github.com/dnaeon/go-vcr/cassette"
+)
+
+type RequestMatcher struct {
+	ignoredHeaders map[string]*string
+}
+
+var ignoredHeaders = map[string]*string{
+	"Date":                   nil,
+	"X-Ms-Date":              nil,
+	"x-ms-date":              nil,
+	"x-ms-client-request-id": nil,
+	"User-Agent":             nil,
+	"Request-Id":             nil,
+	"traceparent":            nil,
+	"Authorization":          nil,
+}
+
+var recordingHeaderMissing = "Test recording headers do not match. Header '%s' is present in request but not in recording."
```

Comment:
I think these all can be constants?

Reply:
they have tokens to indicate which specific value was mismatched.

Reply:
They're just format strings though right?  They can be constants.

Reply:
oh, yes.. sorry, I misunderstood :)